### PR TITLE
Introduces genesis account

### PR DESCRIPTION
**Pull request by eadventurous**
### Description of the Change
Adds genesis account and checks that it is only used to sign txs at genesis.

Genesis account public key is now a necessary config field for every peer.
Genesis account private key is an optional config field, and should be supplied to the peer that signs the genesis tx.

### CODE REVIEW

**Review by MizukiSonoko — APPROVED** [review]

**Review by takemiyamakoto — APPROVED** [review]

File: iroha/src/sumeragi.rs [hunk 1]
```diff
@@ -1489,6 +1489,7 @@ mod tests {
         let mut addresses = Vec::new();
         let mut block_counters = Vec::new();
         let root_key_pair = KeyPair::generate().expect("Failed to generate KeyPair.");
+        let genesis_key_pair = KeyPair::generate().expect("Failed to generate KeyPair.");
```
Comment (takemiyamakoto) [inline iroha/src/sumeragi.rs:1492]: "Failed to generate Genesis KeyPair." might be a better message

File: iroha/src/sumeragi.rs [hunk 2]
```diff
@@ -1615,6 +1618,7 @@ mod tests {
...
```
Comment (takemiyamakoto) [inline iroha/src/sumeragi.rs:1621]: "Failed to generate Genesis KeyPair." might be a better message

File: iroha/src/sumeragi.rs [hunk 3]
```diff
@@ -1765,6 +1771,7 @@ mod tests {
...
```
Comment (takemiyamakoto) [inline iroha/src/sumeragi.rs:1774]: "Failed to generate Genesis KeyPair." might be a better message

File: iroha/src/sumeragi.rs [hunk 4]
```diff
@@ -1925,6 +1934,7 @@ mod tests {
...
```
Comment (takemiyamakoto) [inline iroha/src/sumeragi.rs:1937]: "Failed to generate Genesis KeyPair." might be a better message

File: iroha/src/sumeragi.rs [hunk 5]
```diff
@@ -2082,6 +2094,7 @@ mod tests {
...
```
Comment (takemiyamakoto) [inline iroha/src/sumeragi.rs:2097]: "Failed to generate Genesis KeyPair." might be a better message

File: iroha/src/tx.rs [hunk 6]
```diff
@@ -168,6 +177,7 @@ mod tests {
         let tx = Transaction::new(vec![], AccountId::new("root", "global"), 1000);
         let tx_hash = tx.hash();
         let root_key_pair = &KeyPair::generate().expect("Failed to generate key pair.");
+        let genesis_key_pair = &KeyPair::generate().expect("Failed to generate key pair.");
```
Comment (takemiyamakoto) [inline iroha/src/tx.rs:180]: "Failed to generate Genesis KeyPair." might be a better message

**Review by takemiyamakoto — APPROVED** [review]

**Pull request merged**